typing/typetexp.ml

```diff
@@ -165,9 +165,16 @@ end = struct
   let is_in_scope name =
     TyVarMap.mem name !type_variables
 
-  let add name v =
+  let add ~check loc name v =
     assert (not_generic v);
-    type_variables := TyVarMap.add name v !type_variables
+    let unused = ref check in
+    if check then
+      !Env.add_delayed_check_forward begin fun () ->
+          let warn = Warnings.Unused_type_declaration ("'" ^ name) in
+          if !unused && Warnings.is_active warn
```

Comment:
Testing if the warning is active here is quite imprecise.
I would move this test to `remember_used` to keep track if the warning active when introducing
the alias in order to be able to write:
```ocaml
[@@@warning "-a+34"]
let f (x: int as 'a [@warning "34"]) = ()
```

Reply:
I moved it there